File: src/hello_world_test.py
import pytest
from dataclasses import dataclass

def say_hello(name: str) -> str:
    return f"Hello {name}"

# ------------------------------------------------------------------------------
# TESTS
# ------------------------------------------------------------------------------

@dataclass
class TestData:
    id: str
    input: tuple
    expected: any
    __test__ = False


class TestClass:
    testdata = [
        TestData(id="boy name", input=("Vipin",), expected="Hello Vipin"),
        TestData(id="girl name", input=("Geethu",), expected="Hello Geethu"),
    ]

    @pytest.mark.parametrize(
        "input,expected",
        map(lambda x: (x.input, x.expected), testdata),
        ids=map(lambda x: x.id, testdata),
    )
    def test(self, input, expected):
        actual = say_hello(*input)
        assert actual == expected


if __name__ == "__main__":
    pytest.main()

File: src/helloWorldTest.swift
import Testing

func sayHello(name: String) -> String {
    return "Hello \(name)"
}

struct TestData<Input, Expected> {
    let id: String
    let input: Input
    let expected: Expected
}


@Test(
  "Given different payment type, should display correct alert",
  arguments: [
    TestData(id: "boy name", input: "Vipin", expected: "Hello Vipin"),
    TestData(id: "girl name", input: "Geethu", expected: "Hello Geethu")
  ]
)
func test(testData: TestData<String, String>) {
    let result = sayHello(name: testData.input)
    #expect(result == testData.expected)
}
